### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by GagarinPavel**

### CODE REVIEW

File: PFS/PFS.API/Models/ApplicationContext.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    .WithMany(a => a.Furnitures)
+                    .HasForeignKey(b => b.TypeId)
+                    .HasPrincipalKey(a => a.Id);
+                });
+
+
+            modelBuilder.Entity<Worker>(b =>
+            {
+
+                // Многие ко многим
```
Comment (nantonov) [inline PFS/PFS.API/Models/ApplicationContext.cs:41]: Del and apply formating for this file

File: PFS/PFS.API/Models/Furniture.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace PFS.API.Models
+{
+    public class Furniture
+    {
+        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; set; }
+        public string FurnitureName { get; set; }
```
Comment (nantonov) [inline PFS/PFS.API/Models/Furniture.cs:10]: Just Name

File: PFS/PFS.API/Models/Furniture.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        public int Count { get; set; }
+
+        public int TypeId { get; set; }
```
Comment (nantonov) [inline PFS/PFS.API/Models/Furniture.cs:13]: FurnitureTypeId

File: PFS/PFS.API/Models/FurnitureType .cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace PFS.API.Models
+{
+    public class FurnitureType
+    {
+        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
```
Comment (nantonov) [inline PFS/PFS.API/Models/FurnitureType .cs:8]: Unnecessary

File: PFS/PFS.API/Models/Specialization.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+namespace PFS.API.Models
+{
+    public class Specialization
+    {
+        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
```
Comment (nantonov) [inline PFS/PFS.API/Models/Specialization.cs:7]: Unnecessary

File: PFS/PFS.API/Models/Specialization.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        public int Id { get; set; }
+        public string SpecializationName { get; set; }
+        public List<Worker> Workers { get; set; } = new List<Worker>();
+        [System.Text.Json.Serialization.JsonIgnore]
+        public List<WorkersSpecializations> WorkersSpecializations { get; set; } = new List<WorkersSpecializations>();
```
Comment (nantonov) [inline PFS/PFS.API/Models/Specialization.cs:12]: Why are you ignored this?

File: PFS/PFS.API/Models/Worker.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+namespace PFS.API.Models
+{
+    public class Worker
+    {
+        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
```
Comment (nantonov) [inline PFS/PFS.API/Models/Worker.cs:7]: Unnecessary

File: PFS/PFS.API/Models/Worker.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Surname { get; set; }
+        public DateTime DateОfBirth { get; set; }
+
+        public List<Specialization> Specializations { get; set; }=new List<Specialization>();
+        
+        [System.Text.Json.Serialization.JsonIgnore]
```
Comment (nantonov) [inline PFS/PFS.API/Models/Worker.cs:15]: Unnecessary

File: PFS/PFS.API/Models/ApplicationContext.cs [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+        public DbSet<Specialization> Specializations { get; set; }
+
+        public DbSet<Worker> Workers { get; set; }
+
+        public DbSet<WorkersSpecializations> WorkersSpecializations { get; set; }
+
+           
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (nantonov) [inline PFS/PFS.API/Models/ApplicationContext.cs:25]: Unnecessary at all

**Review by nantonov — CHANGES_REQUESTED** [review]

File: PFS/PFS.API/Models/Specialization.cs (hunk 6 above)
Comment (GagarinPavel) [reply PFS/PFS.API/Models/Specialization.cs:12]: when output this entity there will be an endless loop

File: PFS/PFS.API/Models/Worker.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+namespace PFS.API.Models
+{
+    public class Worker
```
Comment (nantonov) [inline PFS/PFS.API/Models/Worker.cs:5]: Rename to employee

File: PFS/PFS.API/Models/WorkersSpecialization.cs [hunk 11]
```diff
@@ -0,0 +1,11 @@
+﻿namespace PFS.API.Models
+{
+    public class WorkersSpecializations
```
Comment (nantonov) [inline PFS/PFS.API/Models/WorkersSpecialization.cs:3]: Employees not workers

File: PFS/PFS.API/Models/ApplicationContext.cs [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+    public class ApplicationContext : DbContext
+    {
+        public ApplicationContext(DbContextOptions o) : base(o)
+        {
+            Database.EnsureCreated();
+        }
+        public DbSet<Furniture> Furnitures { get; set; }
+        public DbSet<FurnitureType> FurnitureTypes { get; set; }
+        public DbSet<Specialization> Specializations { get; set; }
+        public DbSet<Worker> Workers { get; set; }
```
Comment (nantonov) [inline PFS/PFS.API/Models/ApplicationContext.cs:14]: Employees

**Review by nantonov — CHANGES_REQUESTED** [review]

File: PFS/PFS.API/Models/Specialization.cs [hunk 13]
```diff
@@ -0,0 +1,13 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+namespace PFS.API.Models
+{
+    public class Specialization
+    {
+        public int Id { get; set; }
+        public string SpecializationName { get; set; }
+        public List<Employee> Employees { get; set; } = new List<Employee>();
+        [System.Text.Json.Serialization.JsonIgnore]
```
Comment (nantonov) [inline PFS/PFS.API/Models/Specialization.cs:10]: No needed

**Review by nantonov — CHANGES_REQUESTED** [review]

**Review by nantonov — APPROVED** [review]

**Pull request merged**